libraries/mbed/targets/hal/TARGET_Freescale/TARGET_KPSDK_MCUS/serial_api.c

```diff
@@ -62,16 +62,20 @@ void serial_init(serial_t *obj, PinName tx, PinName rx) {
     UART_HAL_SetStopBitCount(uart_addrs[obj->index], kUartOneStopBit);
     #endif
     UART_HAL_SetBitCountPerChar(uart_addrs[obj->index], kUart8BitsPerChar);
-    UART_HAL_EnableTransmitter(uart_addrs[obj->index]);
-    UART_HAL_EnableReceiver(uart_addrs[obj->index]);
+    UART_HAL_DisableTransmitter(uart_addrs[obj->index]);
+    UART_HAL_DisableReceiver(uart_addrs[obj->index]);
 
     pinmap_pinout(tx, PinMap_UART_TX);
     pinmap_pinout(rx, PinMap_UART_RX);
 
     if (tx != NC) {
+        UART_HAL_FlushTxFifo(uart_addrs[obj->index]);
```

Comment:
+1 for this PR, I had one fix for KSDK on a branch, here's the code 

```
    // Fifo Rx/Tx enabled by default. TX can be enabled/disable only if RE is disabled also
    if (tx != NC) {
        pin_mode(tx, PullUp);
        UART_HAL_SetTxFifoCmd(obj->serial.address, true);
        UART_HAL_FlushTxFifo(obj->serial.address);
        UART_HAL_EnableTransmitter(obj->serial.address);
    }
    if (rx != NC) {
        pin_mode(rx, PullUp);
        UART_HAL_SetRxFifoCmd(obj->serial.address, true);
        UART_HAL_FlushRxFifo(obj->serial.address);
        UART_HAL_EnableReceiver(obj->serial.address);
    }
```
wonder why you flush tx fifo but not rx .